app/jobs/renewal_job.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class RenewalJob < ApplicationJob
+  queue_as :default
+
+  RENEWAL_CUSTOM_MESSAGELIST = {
+    "hatErrorResponse.141": 'Renewal limit reached, cannot be renewed.',
+    "hatErrorResponse.7703": 'Renewal limit reached, cannot be renewed.',
+    "hatErrorResponse.105": 'Item has been recalled, cannot be renewed.',
+    "hatErrorResponse.252": 'Item has holds, cannot be renewed.',
+    "hatErrorResponse.46": 'Item on reserve, cannot be renewed.',
+    "unhandledException": ''
+  }.with_indifferent_access
+
+  def perform(resource:, item_key:, session_token:)
+    symphony_client = SymphonyClient.new
+    redis_client = Redis.new
+    response = symphony_client.renew(
+      resource: resource,
+      item_key: item_key,
+      session_token: session_token
+    )
+
+    case response.status
+    when 200
+      checkout = Checkout.new(JSON.parse(response.body)&.dig('circRecord'))
+
+      redis_client.set("renewal_#{item_key}", {
+        item_key: item_key,
+        result: :success,
+        renewal_count: checkout.renewal_count,
+        due_date: checkout.due_date_human,
+        status: checkout.status_human
+      }.to_json)
+    else
+      Sidekiq.logger.error("renewal_#{item_key}: #{renewal_error_message(response)}")
+      redis_client.set("renewal_#{item_key}", {
+        item_key: item_key,
+        result: :failure,
+        error_message: renewal_error_message(response)
```

Comment:
When the `unhandledException` comes through, then nothing comes through in the logs other than `2020-05-18T22:35:38.273Z pid=60253 tid=ouvxdiup1 class=RenewalJob jid=a4da5f9621eef9a10af1ab5a ERROR: renewal_2089534:3:1:`. I think we need a translation for that error code or we need to dump the response from sirsi here into the logs.

Reply:
I will add a generic msg error for that 👍

Reply:
ok thanks. I was looking at the holds works i did and wondering if I should change that to be the whole raw error message really too, not sure the best approach

Reply:
hmm yeah maybe, depending on if holds error messages is sth for user to see, for renewals they are.

Reply:
Well I was thinking of separating the two: one nice message to users and one raw response from sirsi for the logs which would help us debug quicker.

Reply:
sounds good to me 👍

Reply:
Cool. So you'll just need to update what is logged to be the raw response from sirsi then right? Currently it is still showing the translation

Reply:
this is also done 👍